packages/aws-cdk-lib/aws-cloudfront/lib/realtime-log-config.ts

```diff
@@ -0,0 +1,96 @@
+import { Construct } from 'constructs';
+import { CfnRealtimeLogConfig } from './cloudfront.generated';
+import { IResource, Resource } from '../../core';
+
+/**
+ * Endpoint data streams types.
+ */
+export enum DataStreamType {
+  KINESIS = 'Kinesis'
+}
+
+/**
+ * Contains information about the Amazon Kinesis data stream where you are sending real-time log data.
+ */
+export interface KinesisStreamConfig {
+  /**
+   * The Amazon Resource Name (ARN) of an AWS Identity and Access Management (IAM) role that CloudFront can use to send real-time log data to your Kinesis data stream.
+   */
+  readonly roleArn: string;
+  /**
+   * The Amazon Resource Name (ARN) of the Kinesis data stream where you are sending real-time log data.
+   */
+  readonly streamArn: string;
+}
+
+/**
+ * Contains information about the Amazon Kinesis data stream where you are sending real-time log data in a real-time log configuration.
+ */
+export interface EndPoint {
+  /**
+   * Contains information about the Amazon Kinesis data stream where you are sending real-time log data.
+   */
+  readonly kinesisStreamConfig: KinesisStreamConfig,
+  /**
+   * The type of data stream where you are sending real-time log data.
+   */
+  readonly streamType: DataStreamType;
+}
```

Comment:
I have a few thoughts about this:
1. If the only `streamType` is kinesis, why do we need to have that as a property? Couldn't we just set that for the user rather than having them provide it? I guess my counter thought is will there be more `streamTypes` in the future
2. We recommend not nesting properties in our design guidelines. To me, it doesn't seem like the `kinesisStreamConfig` property is necessary here. Why not just just move the properties from the `kinesisStreamConfig` directly into this interface?

Reply:
> I have a few thoughts about this:
> 
> 1. If the only `streamType` is kinesis, why do we need to have that as a property? Couldn't we just set that for the user rather than having them provide it? I guess my counter thought is will there be more `streamTypes` in the future

Yeah - I wasn't sure if we were going to introduce new types in the future - so wanted to keep it expandable incase

> 2. We recommend not nesting properties in our design guidelines. To me, it doesn't seem like the `kinesisStreamConfig` property is necessary here. Why not just just move the properties from the `kinesisStreamConfig` directly into this interface?

That makes sense to move it inside the interface to adhere to design guidelines

Reply:
Per our slack discussion - it sounds like other stream types will be added so we want to make sure we can expand the implementation when that happens. An enum-like stream config class might be a good approach for this:

```ts
abstract class StreamConfig {
  public static fromKinesisStream(stream: IStream, role: IRole): StreamConfig {
    return new (class extends StreamConfig {
      public _renderStreamConfig() {
        return {
          kinesisStreamConfig: {
            roleArn: role.roleArn,
            streamArn: stream.streamArn,
          },
          streamType: 'Kinesis',
        }
      }
    });
  }
 
  // New streams could just be added here when support is added for them

  private constructor() {}
  
  /*
  * @internal
  */
  public _renderStreamConfig(): any;
  ```